### PR TITLE
Reducing the number of queries for counting # of questions/answers

### LINKED ISSUE #5273: Add the # of asked and answered questions under the questions tab in the profile's page
Hi, this is a first-timers-only issue. This means we've worked to make it more legible to folks who either haven't contributed to our codebase before, or even folks who haven't contributed to open source before.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, consider leaving this one for someone new, and looking through our general help wanted issues. Thanks!

🤔 What you will need to know.
Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

📋 Step by Step
🙋 Claim this issue: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

📝 Update the file app/views/users/profile.html.erb in the plots2 repository. Add the number of asked questions and the number of answered questions under the questions tab in the profile's page.

Replace the lines 73 and 74 : 
`<a href="#asked" class="btn btn-default active" data-toggle="tab">Asked</a>`
 `<a href="#answered" class="btn btn-default" data-toggle="tab">Answered</a>`
by 
`<a href="#asked" class="btn btn-default" data-toggle="tab">Asked<%= "(#{Node.questions.where(status: 1, uid: @profile_user.id).length})" %></a>`
`<a href="#answered" class="btn btn-default" data-toggle="tab">Answered<%= "(#{Answer.where(uid: @profile_user.id).count})" %></a>`

Screen Shot 2019-03-27 at 7 24 53 PM

💾 Commit your changes

🔀 Start a Pull Request. There are two ways how you can start a pull request:

If you are familiar with the terminal or would like to learn it, here is a great tutorial on how to send a pull request using the terminal.

You can also edit files directly in your browser and open a pull request from there.

🏁 Done Ask in comments for a review :)
Please keep us updated
💬⏰ - We encourage contributors to be respectful to the community and provide an update within a week of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance!

If this happens to you, don't sweat it! Grab another open issue.

Is someone else already working on this?
🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- If someone seems stuck, offer them some help! Otherwise, take a look at some other issues you can help with. Thanks!

🤔❓ Questions?
Leave a comment below!

### PR DESCRIPTION
**Pull request by gautamig54**
Fixes #5273  (<=== Add issue number here)

I have reduced the number of queries of counting the number of questions and answers by making a separate variable @count_ans and @count_ques. 
Also, I have added the number of questions asked and Answers posted in the tab on the profile page.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by gautamig54** [comment]
@publiclab/reviewers Kindly review my pull request.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -213,6 +212,8 @@ def profile
           flash.now[:warning] = I18n.t('users_controller.user_has_been_moderated')
         end
       end
+      @count_ans = @profile_user.answers.count
+      @count_ques = @profile_user.questions.length
```
Comment (gauravano) [inline app/controllers/users_controller.rb]: You can use the instance variables defined here - https://github.com/publiclab/plots2/blob/master/app/controllers/users_controller.rb#L143-L188

directly in the view?

Comment (gauravano) [reply app/controllers/users_controller.rb]: Also, the current results are inconsistent - 

Screenshot from 2019-03-31 16-26-14

**Comment by gautamig54** [comment]
@gauravano Kindly review the PR. Thanks!

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -213,6 +212,8 @@ def profile
           flash.now[:warning] = I18n.t('users_controller.user_has_been_moderated')
         end
       end
+      @count_ans = @profile_user.answers.count
+      @count_ques = @profile_user.questions.where(status: 1).length
```
Comment (gauravano) [inline app/controllers/users_controller.rb]: You can remove both of them and can use instance variables defined above in this method. Ref: https://github.com/publiclab/plots2/pull/5310#discussion_r270660582

File: app/views/users/profile.html.erb [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
   <hr />
   <div class="row" id="highlight" style=" text-align: center;">
-    <h3 class="col-md-4 col-xs-12 col-sm-4"><a href = "/tag/question:*/author/<%= params[:id] %>"><%= Node.questions.where(status: 1, uid: @profile_user.id).length %> questions asked </a><br><br> <%= Answer.where(uid: @profile_user.id).count %> answers posted</h3>
+    <h3 class="col-md-4 col-xs-12 col-sm-4"><a href = "/tag/question:*/author/<%= params[:id] %>"><%= @count_ques%> questions asked </a><br><br> <%= @count_ans %> answers posted</h3>
```
Comment (gauravano) [inline app/views/users/profile.html.erb]:
```suggestion
    <h3 class="col-md-4 col-xs-12 col-sm-4"><a href = "/tag/question:*/author/<%= params[:id] %>"><%= @questions.count %> questions asked </a><br><br> <%= @answered_questions.count %> answers posted</h3>
```

Comment (gauravano) [reply app/views/users/profile.html.erb]: Same change below in lines 73 and 74

Comment (gauravano) [reply app/views/users/profile.html.erb]: Thanks!

**Comment by gautamig54** [comment]
I have updated my PR.
Thank you @gauravano for the guidance!

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -2,7 +2,6 @@ class UsersController < ApplicationController
   before_action :require_no_user, only: [:new]
   before_action :require_user, only: %i(edit update save_settings)
   before_action :set_user, only: %i(info followed following followers)
```
Comment (gauravano) [inline app/controllers/users_controller.rb]:
```suggestion
  before_action :set_user, only: %i(info followed following followers)
  
```

File: app/views/users/profile.html.erb [hunk 5]
```diff
@@ -17,7 +17,7 @@
 
   <hr />
   <div class="row" id="highlight" style=" text-align: center;">
-    <h3 class="col-md-4 col-xs-12 col-sm-4"><a href = "/tag/question:*/author/<%= params[:id] %>"><%= Node.questions.where(status: 1, uid: @profile_user.id).length %> questions asked </a><br><br> <%= Answer.where(uid: @profile_user.id).count %> answers posted</h3>
+    <h3 class="col-md-4 col-xs-12 col-sm-4"><a href = "/tag/question:*/author/<%= params[:id] %>"><%= @questions.length%> questions asked </a><br><br> <%= @answered_questions.count %> answers posted</h3>
```
Comment (gauravano) [inline app/views/users/profile.html.erb]:
```suggestion
    <h3 class="col-md-4 col-xs-12 col-sm-4"><a href = "/tag/question:*/author/<%= params[:id] %>"><%= @questions.length %> questions asked </a><br><br> <%= @answered_questions.count %> answers posted</h3>
```

**Comment by gauravano** [comment]
Hey @gautamig54, just a minor change and then it's ready. thanks for being patient here!

**Comment by gautamig54** [comment]
Thank you @gauravano!

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (gauravano) [reply app/controllers/users_controller.rb]: Please commit this one too. Thanks!

Comment (gauravano) [reply app/controllers/users_controller.rb]: @gautamig54 after this one, feel free to mention me on Gitter as this PR is ready.

**Comment by gauravano** [comment]
Merged :tada:. Thanks @gautamig54!

**Pull request merged**